test/jobProcessor.js

```diff
@@ -0,0 +1,130 @@
+import { assert } from 'chai';
+import { default as urlJobProcessor } from '../crawler/urlJobProcessor.js';
+import { default as nock } from 'nock';
+
+let probes;
+
+describe('processUrlJob for a site', () => {
+  beforeEach(() => {
+    probes = {
+      hasHTTPS: false,
+      hasHSTS: false,
+      hasHTTPSRedirect: false,
+      hasManifest: false,
+      hasServiceWorker: false,
+    };
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  context('responding 404 to http and no https', () => {
+    it('should fail on all probes', () => {
+      const site = nock('http://localhost:8000')
+      .get('/')
+      .reply(404);
+
+      return urlJobProcessor.processUrlJob('localhost:8000')
+      .then(ret => {
+        assert.ok(site.isDone());
+        assert.deepEqual(ret, probes);
+      });
+    });
+  });
+
+  context('has no https', () => {
+    it('should fail on all probes', () => {
+      const site = nock('http://localhost:8000')
+      .get('/')
+      .reply(200, '<html></html>');
+
+      return urlJobProcessor.processUrlJob('localhost:8000')
+      .then(ret => {
+        assert.ok(site.isDone());
+        assert.deepEqual(ret, probes);
+      });
+    });
+  });
+
+  context('with http and https response', () => {
+    it('should pass hasHTTPS', () => {
+      nock('http://localhost')
+      .get('/')
+      .reply(200, '<html></html>');
+
+      const site = nock('http://localhost:443')
+      .get('/')
+      .reply(200, '<html></html>');
+
+      return urlJobProcessor.processUrlJob('localhost')
+      .then(ret => {
+        assert.ok(site.isDone());
+        probes.hasHTTPS = true;
```

Comment:
Instead of reusing the same object and changing its properties, it would be cleaner to create a new object every time (so there's no confusion between tests).